src/kokkos/test/kokkos/VertexFinder_t.cc

```diff
@@ -0,0 +1,299 @@
+      Kokkos::deep_copy(ws_d, ws_h);
+
+      std::cout << "M eps, pset " << kk << ' ' << eps << ' ' << (i % 4) << std::endl;
+
+      if ((i % 4) == 0)
+        par = {{eps, 0.02f, 12.0f}};
+      if ((i % 4) == 1)
+        par = {{eps, 0.02f, 9.0f}};
+      if ((i % 4) == 2)
+        par = {{eps, 0.01f, 9.0f}};
+      if ((i % 4) == 3)
+        par = {{0.7f * eps, 0.01f, 9.0f}};
+
+      uint32_t nv = 0;
+      Kokkos::parallel_for(
+          "print", team_policy(1, 1), KOKKOS_LAMBDA(const member_type& team_member) {
+            printf("nt,nv %d %d,%d\n", ws_d(0).ntrks, onGPU_d(0).nvFinal, ws_d(0).nvIntermediate);
+          });
+
+      Kokkos::fence();
+
+#ifdef ONE_KERNEL
+      team_policy policy = team_policy(1, Kokkos::AUTO()).set_scratch_size(0, Kokkos::PerTeam(1024));
+      Kokkos::parallel_for(
+          policy, KOKKOS_LAMBDA(const member_type& team_member) {
+            vertexFinderOneKernel(onGPU_d, ws_d, kk, par[0], par[1], par[2], team_member);
+          });
+#else
+      team_policy policy = team_policy(1, Kokkos::AUTO()).set_scratch_size(0, Kokkos::PerTeam(1024));
+      Kokkos::parallel_for(
+          policy, KOKKOS_LAMBDA(const member_type& team_member) {
+            CLUSTERIZE(onGPU_d, ws_d, kk, par[0], par[1], par[2], team_member);
+          });
+#endif
+      Kokkos::parallel_for(
+          "print", team_policy(1, 1), KOKKOS_LAMBDA(const member_type& team_member) {
+            printf("nt,nv %d %d,%d\n", ws_d(0).ntrks, onGPU_d(0).nvFinal, ws_d(0).nvIntermediate);
+          });
+
+      Kokkos::fence();
+
+      Kokkos::parallel_for(
+          policy,
+          KOKKOS_LAMBDA(const member_type& team_member) { fitVerticesKernel(onGPU_d, ws_d, 50.f, team_member); });
+      // deep copy from device to host
+      Kokkos::deep_copy(onGPU_h, onGPU_d);
+      nv = onGPU_h.data()->nvFinal;
+
+      if (nv == 0) {
+        std::cout << "NO VERTICES???" << std::endl;
+        continue;
+      }
+
+      for (auto j = 0U; j < nv; ++j)
+        if (onGPU_h.data()->ndof[j] > 0)
+          onGPU_h.data()->chi2[j] /= float(onGPU_h.data()->ndof[j]);
+      {
+        auto mx = std::minmax_element(&(onGPU_h.data()->chi2[0]), &(onGPU_h.data()->chi2[nv]));
+        std::cout << "after fit nv, min max chi2 " << nv << " " << *mx.first << ' ' << *mx.second << std::endl;
+      }
+
+      Kokkos::parallel_for(
+          policy,
+          KOKKOS_LAMBDA(const member_type& team_member) { fitVerticesKernel(onGPU_d, ws_d, 50.f, team_member); });
+      Kokkos::deep_copy(onGPU_h, onGPU_d);
+      nv = onGPU_h.data()->nvFinal;
+
+      for (auto j = 0U; j < nv; ++j)
+        if (onGPU_h.data()->ndof[j] > 0)
+          onGPU_h.data()->chi2[j] /= float(onGPU_h.data()->ndof[j]);
+      {
+        auto mx = std::minmax_element(&(onGPU_h.data()->chi2[0]), &(onGPU_h.data()->chi2[nv]));
+        std::cout << "before splitting nv, min max chi2 " << nv << " " << *mx.first << ' ' << *mx.second << std::endl;
+      }
+
+#ifdef KOKKOS_ENABLE_CUDA
```

Comment:
```suggestion
#ifdef KOKKOS_BACKEND_CUDA
```
The `KOKKOS_ENABLE_CUDA` is defined if Kokkos was built with CUDA support (which is done always by the main Makefile). The `KOKKOS_BACKEND_CUDA` is defined by the Makefile only when compiling a file for the CUDA backend (i.e. `KokkosExecSpace` is defined as `Kokkos::Cuda`).